lib/syncUserFollowers.js

```diff
@@ -0,0 +1,124 @@
+/*global Buffer*/
+'use strict';
+
+/**
+ * syncUserFollows API client
+ * Abstraction over the myFT API (v3)
+ * Futher abstration over our myFTClient for sync follower functionality
+ */
+
+const log = require('@financial-times/n-logger').default;
+const config = require('./helpers/config');
+const clientErrors = require('./clientErrors');
+const myFT = require('./myFTClient');
+const kinesis = require('./kinesisClient');
+
+const relationshipProperties = {
+	byTool: config.FT_TOOL_ID,
+	byUser: config.FT_TOOL_ADMIN_ID
+};
+
+const followedProperties = Object.assign({}, relationshipProperties);
+const digestProperties = Object.assign({'type': 'daily', 'timezone': 'Europe/London'}, relationshipProperties);
+
+//const entityProperties = {};// TODO: WIP
+//entityProperties[config.FT_TOOL_DATE_ID] = new Date().toISOString();
+
+const fetchOptions = Object.assign({}, config.fetchOptions);
+fetchOptions.headers = Object.assign({}, fetchOptions.headers, {'X-API-KEY': config.MYFT_API_KEY});
+
+
+/**
+ * Sync user followers
+ * @param {String} groupId - This is likely to be the same of as the licneceId
+ * @param {String} userId -
+ * @returns {Promise} response -
+ */
+module.exports = (groupId, userId) => {
+	const operation = 'syncUserFollowers';
+	return myFT.getConceptsFollowedByGroup(groupId)
+		.catch(err => {
+			// if no concepts are found
+			if (err instanceof clientErrors.NotFoundError) {
+				return [];
+			}
+			throw err;
+		})
+		.then(groupConcepts => {
+			log.silly({operation, groupConcepts: JSON.stringify(groupConcepts)});
+			if (Array.isArray(groupConcepts)) {
+				return groupConcepts;
+			}
+			const msg = 'Group groupConcepts is not an array';
+			log.error({operation, groupConcepts, status: msg});
+			throw new Error(msg);
+		})
+		.then(groupConcepts => {
+			log.silly({operation, groupConceptsLength: groupConcepts.length});
+			const conceptCount = groupConcepts.length;
+			if (conceptCount > 0) {
+				log.silly({operation, subOp: 'groupConceptsFollowed ', userId, groupId, groupConceptsCount: conceptCount});
+				// get the concepts (topics) followed by the user
+				return myFT.getConceptsFollowedByUser(userId)
+					.catch(err => {
+						// if no concepts are found
+						if (err instanceof clientErrors.NotFoundError) {
+							return [];
+						}
+						throw err;
+					})
+					.then(conceptsResp => {
+						if (Array.isArray(conceptsResp)) {
```

Comment:
Will it ever not be an array?

Reply:
Not 100% sure so don't think there is any harm in being defensive here